modular_skyrat/modules/xenos_skyrat_redo/code/xeno_types/queen.dm

```diff
@@ -0,0 +1,97 @@
+	health = 500
+	icon_state = "alienqueen"
+	/// Holds the less lethal tail sweep ability to be granted to the queen later
+	var/datum/action/cooldown/spell/aoe/repulse/xeno/skyrat_tailsweep/hard_throwing/tail_sweep
+	/// Holds the queen screech ability to be granted to her later
+	var/datum/action/cooldown/alien/skyrat/queen_screech/screech
+	melee_damage_lower = 30
+	melee_damage_upper = 35
+
+/mob/living/carbon/alien/humanoid/skyrat/queen/Initialize(mapload)
+	. = ..()
+	tail_sweep = new /datum/action/cooldown/spell/aoe/repulse/xeno/skyrat_tailsweep/hard_throwing()
+	tail_sweep.Grant(src)
+
+	screech = new /datum/action/cooldown/alien/skyrat/queen_screech()
+	screech.Grant(src)
+
+	REMOVE_TRAIT(src, TRAIT_VENTCRAWLER_ALWAYS, INNATE_TRAIT)
+
+	add_movespeed_modifier(/datum/movespeed_modifier/alien_big)
+
+/mob/living/carbon/alien/humanoid/skyrat/queen/Destroy()
+	QDEL_NULL(tail_sweep)
+	QDEL_NULL(screech)
+	return ..()
+
+/mob/living/carbon/alien/humanoid/skyrat/queen/create_internal_organs()
+	internal_organs += new /obj/item/organ/internal/alien/plasmavessel/large/queen
+	internal_organs += new /obj/item/organ/internal/alien/resinspinner
+	internal_organs += new /obj/item/organ/internal/alien/neurotoxin/queen
+	internal_organs += new /obj/item/organ/internal/alien/eggsac
+	..()
+
+/mob/living/carbon/alien/humanoid/skyrat/queen/alien_talk(message, shown_name = name)
+	..(message, shown_name, TRUE)
+
+/obj/item/organ/internal/alien/neurotoxin/queen
+	name = "neurotoxin gland"
+	icon_state = "neurotox"
+	zone = BODY_ZONE_PRECISE_MOUTH
+	slot = ORGAN_SLOT_XENO_NEUROTOXINGLAND
+	actions_types = list(
+		/datum/action/cooldown/alien/acid/skyrat,
+		/datum/action/cooldown/alien/acid/skyrat/lethal,
+		/datum/action/cooldown/alien/acid/corrosion,
+	)
+
+/mob/living/carbon/alien/humanoid/skyrat/queen/death(gibbed)
+	if(stat == DEAD)
+		return
+
+	for(var/mob/living/carbon/carbon_mob in GLOB.alive_mob_list)
+		if(carbon_mob == src)
+			continue
+
+		var/obj/item/organ/internal/alien/hivenode/node = carbon_mob.getorgan(/obj/item/organ/internal/alien/hivenode)
+
+		if(istype(node))
+			node.queen_death()
+
+	return ..()
+
+/datum/action/cooldown/alien/skyrat/queen_screech
+	name = "Deafening Screech"
+	desc = "Let out a screech so deafeningly loud that anything with the ability to hear around you will likely be incapacitated for a short time."
+	button_icon_state = "screech"
+	cooldown_time = 5 MINUTES
+
+/datum/action/cooldown/alien/skyrat/queen_screech/Activate()
+	. = ..()
+	var/mob/living/carbon/alien/humanoid/skyrat/queenie = owner
+	playsound(queenie, 'modular_skyrat/modules/xenos_skyrat_redo/sound/alien_queen_screech.ogg', 100, FALSE, 8, 0.9)
+	queenie.create_shriekwave()
+	shake_camera(owner, 2, 2)
+
+	for(var/mob/living/carbon/human/screech_target in get_hearers_in_view(7, get_turf(queenie)))
+		screech_target.soundbang_act(intensity = 5, stun_pwr = 50, damage_pwr = 10, deafen_pwr = 30) //Only being deaf will save you from the screech
```

Comment:
the queen really doesn't need a AOE stun, considering how viable it is for them to just run off into space everytime a fight goes bad, they should be slightly more vulnerable then the other xenos

Reply:
they already had an AoE stun before, and I assume you can wear earmuffs for full immunity to this